src/app.service.ts

```diff
@@ -0,0 +1,99 @@
+import { Context } from 'koa';
+import { redisGetJson, redisSetJson } from './clients/redis';
+import { LoggerService } from './helpers';
+import { ChannelResult } from './interfaces/channel-result';
+import { CustomerCreditTransferInitiation } from './interfaces/iPain001Transaction';
+import { Channel, NetworkMap } from './interfaces/network-map';
+import { RuleResult } from './interfaces/rule-result';
+import { TypologyResult } from './interfaces/typology-result';
+import apm from 'elastic-apm-node';
+
+export const handleChannels = async (
```

Comment:
I'd like if we can have some better error handling - if ArangoDB query throws an error, then the entire message will crash, and the catch has no context as to what failed. If we can have a try catch around every Integration point, we can do better error logging. We can maybe if ADB errors, at least log the result message to see the transaction outcome, for example.

Reply:
Note: Error mapping. Let's discuss this in a call.